Can't publish universal package after publishing platform-specific package

Publishing vscode-jupyter, the log shows

```
Starting: 🚀 Publish extension
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.212.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /mnt/vss/_work/_temp/709ba5c8-0eb3-4fc4-9d2c-0c4f0c16a08f.sh
npm WARN exec The following package was not found and will be installed: @vscode/vsce@2.20.0
 INFO  Publishing 'ms-toolsai.jupyter (win32-arm64) v2023.7.1002152029'...
 INFO  Extension URL (might take a few minutes): https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter
 INFO  Hub URL: https://marketplace.visualstudio.com/manage/publishers/ms-toolsai/extensions/jupyter/hub
 DONE  Published ms-toolsai.jupyter (win32-arm64) v2023.7.1002152029.
 INFO  Publishing 'ms-toolsai.jupyter (alpine-arm64) v2023.7.1002152029'...
 INFO  Extension URL (might take a few minutes): https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter
 INFO  Hub URL: https://marketplace.visualstudio.com/manage/publishers/ms-toolsai/extensions/jupyter/hub
 DONE  Published ms-toolsai.jupyter (alpine-arm64) v2023.7.1002152029.
 INFO  Publishing 'ms-toolsai.jupyter (darwin-x64) v2023.7.1002152029'...
 INFO  Extension URL (might take a few minutes): https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter
 INFO  Hub URL: https://marketplace.visualstudio.com/manage/publishers/ms-toolsai/extensions/jupyter/hub
 DONE  Published ms-toolsai.jupyter (darwin-x64) v2023.7.1002152029.
 INFO  Publishing 'ms-toolsai.jupyter (linux-arm64) v2023.7.1002152029'...
 INFO  Extension URL (might take a few minutes): https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter
 INFO  Hub URL: https://marketplace.visualstudio.com/manage/publishers/ms-toolsai/extensions/jupyter/hub
 DONE  Published ms-toolsai.jupyter (linux-arm64) v2023.7.1002152029.
 INFO  Publishing 'ms-toolsai.jupyter (win32-x64) v2023.7.1002152029'...
 INFO  Extension URL (might take a few minutes): https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter
 INFO  Hub URL: https://marketplace.visualstudio.com/manage/publishers/ms-toolsai/extensions/jupyter/hub
 DONE  Published ms-toolsai.jupyter (win32-x64) v2023.7.1002152029.
 INFO  Publishing 'ms-toolsai.jupyter v2023.7.1002152029'...
 ERROR  ms-toolsai.jupyter v2023.7.1002152029 already exists.
##[error]Bash exited with code '1'.
Finishing: 🚀 Publish extension
```

My interpretation is that it's failing to publish the generic package because the platform-specific package was published with the same version number. I think this could be from https://github.com/microsoft/vscode-vsce/pull/873

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "@vscode/vsce",
-	"version": "2.20.0",
+	"version": "2.20.1",
 	"description": "VSCode Extension Manager",
 	"repository": {
 		"type": "git",

--- a/src/publish.ts
+++ b/src/publish.ts
@@ -175,7 +175,7 @@ async function _publish(packagePath: string, manifest: Manifest, options: IInter
 		if (extension && extension.versions) {
 			const versionExists = extension.versions.some(v =>
 				(v.version === manifest.version) &&
-				(options.target ? v.targetPlatform === options.target : true));
+				(v.targetPlatform === options.target));
 
 			if (versionExists) {
 				if (options.skipDuplicate) {

